toolkit/tools/pkgworker/pkgworker.go

```diff
@@ -177,9 +179,23 @@ func buildSRPMInChroot(chrootDir, rpmDirPath, toolchainDirPath, workerTar, srpmF
 		return
 	}
 
-	err = chroot.Run(func() (err error) {
-		return buildRPMFromSRPMInChroot(srpmFileInChroot, outArch, runCheck, defines, packagesToInstall, useCcache)
-	})
+	// Run the build in a go routine so we can monitor and kill it if it takes too long.
+	results := make(chan error)
+	go func() {
+		buildErr := chroot.Run(func() (err error) {
+			return buildRPMFromSRPMInChroot(srpmFileInChroot, outArch, runCheck, defines, packagesToInstall, useCcache)
+		})
+		results <- buildErr
+	}()
+
+	select {
+	case err = <-results:
+	case <-time.After(timeout):
+		shell.PermanentlyStopAllProcesses(unix.SIGKILL)
```

Comment:
Do you know which processed this kills? I'm guessing it's only the children, but in that case, could we re-name it to something like `PermanentlyStopAllChildrenProcesses` just to be more clear?

Reply:
Specifically, it sends the signal to everything tracked by `shell.activeCommands` which is a map of running commands spawned by the `shell` package.

I'm fine with updating the name.

Reply:
Done